mitodl/mit-learn
# fix edx file imports

### What are the relevant tickets?
closes https://github.com/mitodl/hq/issues/6031

### Description (What does it do?)
This pr removes duplicate edx content files and cleans up content file content for subtitle files.  We were creating duplicates of edx content files because both the code that was intended to find existing documents to update them instead of creating new ones and the code that unpublished and deleted documents that were no longer in the course folder had bugs.

It also populates the file path and file extension for edx content files and does not attempt to extract content with tilka for non-text files such as csv files or various code file formats.

### How can this be tested?
If you don't already have some xpro files in your database run ` docker-compose run web ./manage.py backpopulate_xpro_data`,  `docker-compose run web ./manage.py backpopulate_xpro_files in the main branch

On this branch

Run `docker-compose run web ./manage.py backpopulate_xpro_data` , `docker-compose run web ./manage.py backpopulate_xpro_files --overwrite`
Go to http://api.open.odl.local:8063/api/v1/contentfiles/?platform=xpro and verify that you have 10315 files
From the shell verify that
 ContentFile.objects.filter(run__learning_resource__platform__code='xpro' ).filter(file_extension__isnull=True).count() is 0

Find a srt file useing

ContentFile.objects.filter(run__learning_resource__platform__code='xpro').filter(file_extension='.srt').first()
Verify that the content field does not have timestamps
